### PR TITLE
Log improvement of multi-cloud

### PR DESCRIPTION
**Pull request by ThisIsClark**
**What this PR does / why we need it**:
Improve the log module of multi-cloud
Log file path, log level and log format can be configured in log.conf.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

**What this log module provided**
Log file path, level and format can be configured in 

> log.conf

 which is in opensds's root path, configuration is like this:
```
path = /var/log/multi-cloud
level = info
format = [%time%] [%level%] [%filename%] [%funcName%():%lineNo%] [PID:%process%] %message%
```
`path` indicates where the log file is stored.
`level` indicates the level of logger.
`format` indicates the format of log. `%time%`, `%level%`, `%filename%`, `%funcName%`, `%lineNo%`,  `%process%` and `%message%` can be used to configure to format
And if path, level or format are not configred, the default path is `/var/log/opensds`, the default level is `info`, and the default format is `[%time%] [%level%] [%filename%] [%funcName%():%lineNo%] [PID:%process%] %message%`

**How to write log**
First of all, you need to import the package:
```
import (
    log "github.com/sirupsen/logrus"
)
```
The log is divided into four levels: debug, info, warn, error
And you can use these four function to write the log: `log.Debug`, `log.Info`, `log.Warn`, `log.Error`

**Test Report**
```
root@ubuntu:/var/log/multi-cloud # ls
api.7ac3a9e8f06f.root.log.ERROR.2019092315.1
api.7ac3a9e8f06f.root.log.INFO.2019092315.1
api.ERROR
api.INFO
dataflow.ERROR
dataflow.f0034acf9006.root.log.ERROR.2019092315.1
dataflow.f0034acf9006.root.log.INFO.2019092315.1
dataflow.INFO
datamover.655a128faf51.root.log.INFO.2019092315.1
datamover.INFO
s3.5a1bed9919eb.root.log.INFO.2019092315.1
s3.INFO
```
```
root@ubuntu:/var/log/multi-cloud # cat api.ERROR
[2019-09-23T15:53:56+08:00] [ERROR] [/root/go/src/github.com/opensds/multi-cloud/api/pkg/filters/auth/keystone.go] [github.com/opensds/multi-cloud/api/pkg/filters/auth.NewKeystone():45] [PID:1] When get auth client:%!(EXTRA *url.Error=Get http://10.10.3.154/identity/: dial tcp 10.10.3.154:80: connect: no route to host)
```

### CODE REVIEW

File: api/pkg/backend/service.go [hunk 1]
```diff
@@ -56,12 +56,12 @@ func (s *APIService) GetBackend(request *restful.Request, response *restful.Resp
...
 		return
 	}
-	log.Logf("Received request for backend details: %s\n", request.PathParameter("id"))
+	log.Infof("Received request for backend details: %s\n", request.PathParameter("id"))
 	id := request.PathParameter("id")
 	ctx := context.Background()
 	res, err := s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: id})
 	if err != nil {
-		log.Logf("failed to get backend details: %v\n", err)
+		log.Infof("failed to get backend details: %v\n", err)
```
Comment (sfzeng) [inline api/pkg/backend/service.go]: It's better to use the level of error instead of info here .

File: api/pkg/backend/service.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func (s *APIService) listBackendDefault(request *restful.Request, response *rest
 
 	limit, offset, err := common.GetPaginationParam(request)
 	if err != nil {
-		log.Logf("get pagination parameters failed: %v\n", err)
+		log.Infof("get pagination parameters failed: %v\n", err)
```
Comment (sfzeng) [inline api/pkg/backend/service.go]: same here

**Comment by sfzeng** [comment]
1. Please fix the conflict.
2. For all logs referred to failed or error, please use log.Errorf instead of log.Infof.
Thanks.

**Comment by sfzeng** [comment]
Hi @ThisIsClark , can we add file name and function name to the log? That will be very helpful for debugging.

**Comment by ThisIsClark** [comment]
> Hi @ThisIsClark , can we add file name and function name to the log? That will be very helpful for debugging.

Hi @sfzeng , file name, function name and line number have been added to the log. You can get more details in the release note of this PR.

**Comment by ThisIsClark** [comment]
> 1. Please fix the conflict.
> 2. For all logs referred to failed or error, please use log.Errorf instead of log.Infof.
>    Thanks.

Done

**Comment by sfzeng** [comment]
LGTM

**Comment by wisererik** [comment]
please fix conflicts @ThisIsClark

**Comment by sfzeng** [comment]
LGTM

**Pull request merged**